File: src/ru/greeneyes/project/pomidoro/UIBundle.java
package ru.greeneyes.project.pomidoro;

import com.intellij.CommonBundle;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.PropertyKey;

import java.lang.ref.Reference;
import java.lang.ref.SoftReference;
import java.util.ResourceBundle;

/**
 * Copied from {@link com.intellij.ui.UIBundle}.
 *
 * User: dima
 * Date: Jun 1, 2010
 */
public class UIBundle {
	private static Reference<ResourceBundle> ourBundle;
	@NonNls
	protected static final String PATH_TO_BUNDLE = "resources.messages";

	public static String message(@PropertyKey(resourceBundle = "resources.messages") String key, Object... params) {
		return CommonBundle.message(getBundle(), key, params);
	}

	private static ResourceBundle getBundle() {
		ResourceBundle bundle = null;
		if (ourBundle != null) bundle = ourBundle.get();
		if (bundle == null) {
			bundle = ResourceBundle.getBundle(PATH_TO_BUNDLE);
			ourBundle = new SoftReference<ResourceBundle>(bundle);
		}
		return bundle;
	}

}

File: src/pomodoro/UIBundle.kt
package pomodoro

import com.intellij.AbstractBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey
import java.lang.ref.Reference
import java.lang.ref.SoftReference
import java.util.*

/**
 * Copied from [com.intellij.ui.UIBundle].
 */
object UIBundle {
    private var ourBundle: Reference<ResourceBundle>? = null
    @NonNls private const val PATH_TO_BUNDLE = "messages"

    @JvmStatic fun message(@PropertyKey(resourceBundle = "messages") key: String, vararg params: Any): String =
        AbstractBundle.message(bundle, key, *params)

    private val bundle: ResourceBundle
        get() {
            var bundle: ResourceBundle? = null
            if (ourBundle != null) bundle = ourBundle!!.get()
            if (bundle == null) {
                bundle = ResourceBundle.getBundle(PATH_TO_BUNDLE)
                ourBundle = SoftReference(bundle)
            }
            return bundle!!
        }
}
